### ISSUE TITLE
Implement secondary texts

### ISSUE THREAD

**Issue by abartov** [abartov/bybeconv#174]
Per the bullet on secondary texts in [the design for heterogeneous collections](/wiki/Heterogeneous-Collection-Design-Proposal), we need a new boolean flag in Works specifying whether they are primary (default) or secondary texts.  Primary texts are poems, essays, stories, novels, etc., whereas secondary texts are forewords, publisher's notes, brief biographies preceding texts, etc.

This distinction should have (at least) the following implications:

1. When editing "htmlfile" markdown for publication (edit_markdown view), editors should be allowed to specify, *per text*, whether it is primary or secondary.  Primary being the default, we just need some magic-word syntax to mark texts as secondary. Remember that edit_markdown allows editing a single markdown buffer that would be *split* into multiple eventual WEM entities, so the designation of a unit as secondary must somehow be in the form of markdown magic-words, just like the &&& separator itself. Perhaps an optional suffix to that line would be clearest. I.e. &&& <title of work> %%% -- where the presence of %%% marks the following markdown unit (whose title is given in this line) as secondary.
2. At entity creation (create_WEM_new), the secondary field should be assigned, per the presence of the magic-word (%%% above)
3. _browsing_ actions such as /works, including various filterings, as well as _random_ fetches (surprise_work etc.), and even the total work count, should _ignore_ (filter out) all secondary texts.  Secondary texts would only be discoverable by either full-text search (they _should_ be indexed!) or by direct link from a TOC page (currently only author's TOC page, but eventually there would be periodical TOC pages too).

Let me know if you have questions or need more details to implement this.

**Comment by damisul** [comment]
@abartov , can you briefly describe me purpose of html files model and how it is used in project? Also what it WEM you've mentioned there.

**Comment by damisul** [comment]
_3. browsing actions such as /works, including various filterings, as well as random fetches (surprise_work etc.), and even the total work count, should ignore (filter out) all secondary texts. Secondary texts would only be discoverable by either full-text search (they should be indexed!) or by direct link from a TOC page (currently only author's TOC page, but eventually there would be periodical TOC pages too)._

What about API calls?

**Comment by abartov** [comment]
HtmlFile _used_ to be an entity representing the legacy Word-generated HTML files that benyehuda.org used to consist of. (The project is still called bybeconv because it started as a project to convert those to a proper DB-backed system.)  There are still remnants of parsing code for those legacy files there, even though we have been done with the conversion for a while now.

But materials are still being _prepared_ for the site in Word files, and need to be converted (using pandoc + human editing), and the HtmlFile entity has been repurposed to express that interim stage. 

In other words, the way new works enter the library is by editors creating new 'HtmlFile' entities with attached DOCX files, feeding metadata and editing the pandoc-converted markdown (including, potentially, marking where to separate individual works in a collection of poetry or essays in a larger DOCX file), and then hitting the "frbrize" button, which eventually called the HtmlFile.create_WEM_new method, which creates the WEM (Work, Expression, Manifestation) entities and ends the HtmlFile's role.

**Comment by damisul** [comment]
@abartov , I started working on this last summer, but later we switched to Sara project and work halted.
There is some work done, but it is far from being finished.
I've pushed my changes to https://github.com/abartov/bybeconv/tree/damir_secondary_works

**Comment by abartov** [comment]
DB and search changes from the Feb 2023 branch have been merged in production today.

The upload form is being significantly re-designed, so the secondary-text marking will be part of the new UX/design, so I'm closing this issue as done.
